OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -89,59 +100,27 @@ public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool qu
 
 		public void ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString != "DeployToUpgrade")
+			if (order.OrderString != "DeployToUpgrade" || deployState == DeployState.Deploying)
 				return;
 
 			if (!IsOnValidTerrain())
 				return;
 
-			if (isUpgraded)
+			if (!order.Queued)
+				self.CancelActivity();
```

Comment:
```
(22:40:26) antares79: pchote: I think the queueing not working properly on #10014 is because the Move activity is doing freaky shit again
(22:41:07) antares79: at the point where my breakpoint stops, there's not NextActivty on MoveSecondHalf, although there are three more waypoints the unit is supposed to go to
(22:48:44) antares79: pchote: confirmed: [see image]
(22:49:16) antares79: the turn and callfunc are from DTU, the two moves are me ordering the thumper around before deploying
```
![move-activity-madness](https://cloud.githubusercontent.com/assets/4331210/11195879/aa4230f6-8cb4-11e5-8059-6f001d3e712c.png)